src/groupByKey.ts

```diff
@@ -1,7 +1,30 @@
+/* eslint-disable no-restricted-syntax */
 type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const arrOfValues: string[] | number[] = [];
+  const result: GroupsMap<T> = {};
+
+  // eslint-disable-next-line no-restricted-syntax
```

Comment:
I have no idea how to avoid using loop hear ( 8:3  error  iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations  no-restricted-syntax)